python/ml4ir/model/ranking_model.py

```diff
@@ -416,6 +417,16 @@ def _predict_score(features, label):
                             "{} was not found in input training data".format(feature_name)
                         )
 
+                # Explode context features to each record for logging
```

Comment:
"for logging" only?

Reply:
So these are metadata fields mostly. But even for things like query text, we explode it here so that we can log. The actual LSTM would be trained in the feature layer on a single context feature per query and then tiled for each record in that query.

Reply:
You are looking at the inference code here. Where I need to convert each TFRecord back to a log-able CSV(or a pandas dataframe) with one record per row.